### PR TITLE
[backport -> release/3.4.x] chore(deps): bump openresty from 1.21.4.1 to 1.21.4.3

### PR DESCRIPTION
**Pull request by samugi**
### Summary

backport of:

- https://github.com/Kong/kong/pull/11360
- https://github.com/Kong/kong/pull/11622
- https://github.com/Kong/kong/pull/11952

to release/3.4.x

In order to enable upgrading lua-kong-nginx-module to 0.8.0 and backport the request_id feature

related PR (EE): https://github.com/Kong/kong-ee/pull/7206

### Checklist

- [x] (no) The Pull Request has tests
- [x] (no) A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/README.md)
- [x] (no) There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Full changelog

* [Implement ...]

### Issue reference

KAG-3040

### CODE REVIEW

**Comment by chronolaw** [comment]
Should we upgrade to 1.21.4.3?

**Comment by samugi** [comment]
> Should we upgrade to 1.21.4.3?

yes, I'm doing this one step at a time to make troubleshooting easier if any tests start failing

**Comment by tzssangglass** [comment]
just curious, I noticed this PR deleted some patches, it has been included in the new version of OpenResty?

**Comment by samugi** [comment]
yes @tzssangglass you can see this for example in: https://github.com/Kong/kong/pull/11952 (that this PR is backporting)

**Review by bungle — APPROVED** [review]

File: changelog/unreleased/kong/bump-openresty-1.21.4.3.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+message: "Bumped OpenResty from 1.21.4.2 to 1.21.4.3"
```
Comment (dndx) [inline changelog/unreleased/kong/bump-openresty-1.21.4.3.yml:1]: Keep just one changelog entry of bump from `1.21.4.1` to `1.21.4.3`, not two separate ones.

**Review by dndx — DISMISSED** [review]

File: changelog/unreleased/kong/bump-openresty-1.21.4.3.yml (hunk 1 above)
Comment (samugi) [reply changelog/unreleased/kong/bump-openresty-1.21.4.3.yml]: done, thanks

**Pull request merged**